### PR TITLE
Add record command

### PR DESCRIPTION
**Pull request by lblackstone**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,6 +67,8 @@ GO111MODULE=on go install
 -   `trace <kind> [<namespace>/]<name>`, which "traces" the changes a complex Kubernetes resource
     makes throughout a cluster, and aggregates them into a high-level summary, which is updated in
     real time.
+-   `record <apiVersion> <kind> [<namespace>/]<name>`, which in real time emits all changes to any
+    field in a Kubernetes resource, as a JSON array.
```
Comment (hausdorff) [inline README.md]: It emits the changes (_e.g._, a diff) or it emits the entire object any time there is a change?

File: cmd/record.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		events, err := watch.Forever(args[0], args[1], watch.ThisObject(namespace, name))
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		fmt.Print("[\n  ")
```
Comment (hausdorff) [inline cmd/record.go]: Something is weird about the first part of this, causing an object to get printed with a bit of extra space:

```
[
    {
  }
]
```

File: cmd/record.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+						}
+						fmt.Print("  ")
+						if output, err := json.MarshalIndent(o.Object, "  ", "  "); err != nil {
+							log.Fatal(err)
+						} else {
+							fmt.Print(string(output))
+						}
+					}
+				case apiwatch.Deleted:
+                    // Nothing to print.
```
Comment (hausdorff) [inline cmd/record.go]: Why spaces instead of tabs here?

**Comment by hausdorff** [comment]
Also, a smaller thing. When you press ctrl-c you still get the `^C` character at the end:

```
  }^C
]
```

Also did we test this on windows?

File: README.md (hunk 1 above)
Comment (lblackstone) [reply README.md]: The entire object. See https://github.com/pulumi/pulumi-kubernetes/blob/2970ee20fd92337f643f716aa90d8e4b17d236d6/pkg/await/recordings/podCreateSuccess.json as an example I recorded.

File: cmd/record.go (hunk 2 above)
Comment (lblackstone) [reply cmd/record.go]: I'll double-check.

File: cmd/record.go (hunk 3 above)
Comment (lblackstone) [reply cmd/record.go]: Apparently forgot to run `gofmt`. Fixed.

File: cmd/record.go (hunk 2 above)
Comment (lblackstone) [reply cmd/record.go]: Good catch; fixed.

**Comment by lblackstone** [comment]
@hausdorff I did not test on Windows. I was just redirecting the output of the command to a file, so I guess that avoided the `^C` ending up in the output.

**Pull request merged**